src/main/java/com/ait/lienzo/client/core/types/BoundingPoints.java

```diff
@@ -35,13 +37,40 @@ public BoundingPoints(final BoundingBox bbox)
 
         double h = bbox.getHeight();
 
-        m_array.push(new Point2D(x + 0, y + 0));
+        double x0;
+        double y0;
+        double x1;
+        double y1;
+
+        if (bbox.getMaxX() > bbox.getMinX())
```

Comment:
well @hasys I think the current approach is fine... I mean the BoundingBox class handles just 4 points (coordinates) internally, and the intersect method uses these absolute coordinates, so that seems fine, it does not understand about direction, there is no need for... i'ts just an state object.
On the other hand the BoundingPoints class takes care of the direction, and its state is an array of points. So it depends on the way you need to obtain the points and the goal you want to achieve... but IMO is fine having as two separate classes.